managed_config/index.bs

```diff
@@ -0,0 +1,116 @@
+
+## Managed configuration ## {#managed-configuration}
+
+For the purposes of this specification the device administrator can control a list of managed web applications. Each entry in this list can have a set JSON configuration, which is accessible by the documents hosted at the origin of the web application.
+
+Note that this does not apply to anonymous contexts, where the observed state is indistinguishable from a non-managed one.
+
+## <dfn dfn>Data model</dfn> ## {#data-model-section}
+
+We can assume without loss of generality that managed configuration is stored in a two-level key-value store.
+
+[=Data model=] is declared as a [=map=] [=map=] where each [=key=] is a configured [=origin=] and the value is a <xmp>record<DOMString, object></xmp> with items corresponding to per-origin configuration keys and their values.
+
+## Data integrity verification ## {#data-integrity-verification}
+
+There is no managed configuration verification mechanism defined in the specification, which implies that the website should assume that the configuration could be tampered with, stolen or be replayed on a non-managed device. 
+
+Alternative security measures should be put in place by the website itself.
+
+# Extensions to the {{Navigator}} interface # {#navigator-extensions}
+
+<xmp class="idl">
+[
+  SecureContext
+] partial interface Navigator {
+  [SecureContext, SameObject]
+  readonly attribute NavigatorManagedData managed;
+};
+</xmp>
+
+
+{{managed}} attribute
+When getting, the managed attribute always returns the same instance of the {{NavigatorManagedData}} object.
+
+# {{NavigatorManagedData}} interface # {#navigatormanageddata-interface}
+
+<xmp class="idl">
+[
+  SecureContext,
+  Exposed=Window
+] interface NavigatorManagedData : EventTarget {
+  // Managed Configuration API.
+  Promise<record<DOMString, object>> getManagedConfiguration(sequence<DOMString> keys);
+  attribute EventHandler onmanagedconfigurationchange;
+};
+</xmp>
+
+Methods on this interface typically complete asynchronously, queuing work on the <dfn>managed data task source</dfn>.
+
+## {{getManagedConfiguration()}} method ## {#getmanagedconfiguration-method}
+
+  The {{getManagedConfiguration()}} method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. Let |map| be the [=data model=] configured by the device administrator.
+      1. Let |origin| be [=global object=]'s [=origin=].
+      
+          NOTE: Third-party contexts have access to the same configuration values as if they were the top-level document.
+      1. If there is no entry in the |map| with key equal to |origin|, reject |promise| with a {{NotAllowedError}} DOMException.
+      1. Create an empty IDL [=record=] |record|.
+      1. For each |key| of the {{keys}}:
+          1. If there is a record with key |key| in |map|[|origin|]:
+              1. Add an attribute to |record| with |key| as the key and |map|[|origin|][|key|] as the value.
```

Comment:
Since there is just a single step at each level of nesting here I would just use commas to make this a single sentence:

> For each key of the keys, if there is a record with key key in map[origin], add an attribute to record with key as the key and map[origin][key] as the value.